web/src/navbar_menus.js

```diff
@@ -13,45 +14,61 @@ export function is_navbar_menus_displayed() {
 }
 
 export function handle_keyboard_events(event_name) {
-    // We don't need to process arrow keys in navbar menus for spectators
-    // since they only have gear menu present.
-    if (
-        popover_menus.is_personal_menu_popover_displayed() &&
-        (event_name === "left_arrow" || event_name === "gear_menu") &&
-        !page_params.is_spectator
-    ) {
-        // Open gear menu popover on left arrow.
-        personal_menu_popover.toggle();
-        gear_menu.toggle();
-        return true;
+    const allowed_events = new Set(["gear_menu", "left_arrow", "right_arrow"]);
+    if (!allowed_events.has(event_name)) {
+        return false;
     }
```

Comment:
For example, a first commit could put this check at the top. Another could introduce `change_active_navbar_menu` and start calling it rather than calling two `toggle` calls. etc.